src/output-tx.c

```diff
@@ -138,35 +148,87 @@ static TmEcode OutputTxLog(ThreadVars *tv, Packet *p, void *thread_data, PacketQ
         if (!(AppLayerParserStateIssetFlag(f->alparser, APP_LAYER_PARSER_EOF)))
         {
             int tx_progress = AppLayerParserGetStateProgress(p->proto, alproto,
-                                                             tx, FlowGetDisruptionFlags(f, STREAM_TOSERVER));
-            if (tx_progress < tx_progress_done_value_ts) {
-                SCLogDebug("progress not far enough, not logging");
-                break;
+                tx, FlowGetDisruptionFlags(f, STREAM_TOSERVER));
+            if (tx_progress == tx_progress_done_value_ts) {
+                ts_ready = 1;
             }
 
             tx_progress = AppLayerParserGetStateProgress(p->proto, alproto,
-                                                         tx, FlowGetDisruptionFlags(f, STREAM_TOCLIENT));
-            if (tx_progress < tx_progress_done_value_tc) {
-                SCLogDebug("progress not far enough, not logging");
-                break;
+                tx, FlowGetDisruptionFlags(f, STREAM_TOCLIENT));
+            if (tx_progress == tx_progress_done_value_tc) {
+                tc_ready = 1;
             }
         }
 
+        if (!(ts_ready || tc_ready)) {
+            SCLogNotice("progress not for enough, not logging");
+            break;
+        }
+
+        /* Prevent the processing of one direction getting farther
```

Comment:
I think we're both after the same thing, but expanding the scope of what I'm trying to attempt in this fix.

For logging we have request complete, and response complete. So tx complete is request_complete && response_complete.  It would be nice to log the requests as they are seen, prior to the responses, but we don't do that now, and I need to retain that behaviour until we can fix the scenario of overlapping transactions, which is another task.  But this gets us half way there by within the scope of transaction, logging each part on receipt if requested to do so.